kaiaai_telemetry_node:
  ros__parameters:

    laser_scan:
      mask_radius_meters: 0.1
      topic_name_pub: "scan"
      frame_id: "base_scan"
#      lds_model: "LDS02RR"
#      lds_model: "YDLIDAR-X4"

    laser_sensor:
      model:            ['YDLIDAR-X4','XIAOMI-LDS02RR','YDLIDAR-X2-X2L','YDLIDAR-X3-PRO','YDLIDAR-X3-PRO','YDLIDAR-X3','NEATO-XV11','RPLIDAR-A1','DELTA-2G']
      angle_offset_deg: [         0.0,          -180.0,             0.0,             0.0,             0.0,         0.0,      -180.0,      -180.0,     -90.0]
      pub_scan_size:    [         720,             360,             720,             720,             720,         720,         360,         720,       360]
      clockwise:        [        true,            true,            true,            true,            true,        true,       false,        true,      true]
      range_min_meters: [        0.15,            0.15,             0.1,            0.12,            0.12,        0.12,        0.15,        0.15,      0.15]
      range_max_meters: [        12.0,             6.0,             8.0,             8.0,             8.0,         8.0,         6.0,        12.0,       5.0]
      # 'HLS-LFCD2', 'XIAOMI-LDS01RR', 'LDROBOT-LD14P', 'LDROBOT-LD20', 'DELTA-2A'
      #         0.0,              0.0,
      #         360,              360,
      #        true,             true, 
      #        0.12,             0.15,             0.1
      #         3.5,              9.0,             8.0

    telemetry:
      topic_name_sub: "telemetry"

    tf:
      frame_id: "odom"
      child_frame_id: "base_footprint"

    joints:
      topic_name_pub: "joint_states"
      wheel:
        right: "wheel_right_joint"
        left: "wheel_left_joint"

    odometry:
      frame_id: "odom"
      child_frame_id: "base_footprint"
      topic_name_pub: "odom"
